### PR TITLE
Upgrading to AWS provider 3.x

### LINKED ISSUE #65: Module requires aws provider version 3.1.0
[versions.tf](https://github.com/cloudposse/terraform-aws-ecr/blob/master/versions.tf#L5) requires aws = ">= 2.34"
But encryption_configuration in [main.tf](https://github.com/cloudposse/terraform-aws-ecr/blob/master/main.tf#L17) was introduced with aws provider version [3.1.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#310-august-07-2020)

### PR DESCRIPTION
**Pull request by jamengual**
## what
* upgrade aws provider to > 3.1

## why
* version 3.1 introduced a bug in encryption configs

## references
* closes #65

### CODE REVIEW

File: versions.tf [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
-  required_version = ">= 0.12.0"
-
+  required_version = ">= 0.13.3"
   required_providers {
-    aws      = ">= 2.34"
-    template = ">= 2.0"
-    local    = ">= 1.2"
+    aws = {
+      source = "hashicorp/aws"
+      version       = ">= 3.0, < 4.0"
```
Comment (aknysh) [inline versions.tf:6]:
```suggestion
      version       = ">= 3.1"
```

Comment (aknysh) [reply versions.tf:6]: also do the same for `examples/complete/versions.tf`

File: versions.tf [hunk 2]
```diff
@@ -1,9 +1,17 @@
...
+      version = ">= 3.1"
+    }
+    template = {
+      source  = "hashicorp/template"
+      version = ">= 2.0"
+    }
+    local = {
+      source  = "hashicorp/local"
+      version = ">= 1.3"
+    }
```
Comment (Nuru) [inline versions.tf:15]: Preserve Terraform 0.12 compatibility. I also do not see a reason to disallow `local` version 1.2
```suggestion
    aws      = ">= 3.1"
    template = ">= 2.0"
    local    = ">= 1.2"
```

**Review by Nuru — CHANGES_REQUESTED** [review]
@aknysh I would like to preserve as much backward compatibility as possible, so I think we should limit this PR to just a change to the `aws` provider version constraint. Users of the module are free to use higher versions if they want to, but should not be forced to. Do you agree?

File: versions.tf [hunk 3]
```diff
@@ -1,9 +1,17 @@
 terraform {
-  required_version = ">= 0.12.0"
-
+  required_version = ">= 0.13.3"
```
[the remark this answers is not in the document]
Comment (jamengual) [reply versions.tf:2]: Users should pin versions I think.

File: versions.tf (hunk 2 above)
Comment (jamengual) [reply versions.tf:15]: I just updated to the latest version, does it have to be 1.2?

Comment (aknysh) [reply versions.tf:15]: let's keep 1.2

File: versions.tf (hunk 3 above)
[the remark this answers is not in the document]
Comment (aknysh) [reply versions.tf:2]: yes. we should use TF 0.12 compatibility.
No reason to restrict to 0.13 if we need only the `aws` provider v3.1 for one of the features added to the module.

In any case, we need to try to not restrict to the latest versions unnecessary.

[the remark this answers is not in the document]
Comment (jamengual) [reply versions.tf:2]: should we use ">= 0.12.0" or ">= 0.12" ?

**Review by aknysh — APPROVED** [review]
thanks @jamengual

**Pull request merged**